packages/mui-base/src/useAutocomplete/useAutocomplete.js

```diff
@@ -1000,15 +1000,14 @@ export default function useAutocomplete(props) {
     }
   };
 
-  // Prevent input blur when interacting with the combobox
-  const handleMouseDown = (event) => {
-    if (event.target.getAttribute('id') !== id) {
-      event.preventDefault();
+  const handleInputMouseDown = (event) => {
+    if (inputValue === '' || !open) {
+      handlePopupIndicator(event);
     }
   };
 
   // Focus the input when interacting with the combobox
-  const handleClick = () => {
+  const handleClick = (event) => {
     inputRef.current.focus();
```

Comment:
To see if it makes sense to use this as an opportunity to fix this line. Focus on click no sense per https://github.com/mui/material-ui/issues/29381#issuecomment-1442594854